hs_core/hydroshare/hs_bagit.py

```diff
@@ -204,19 +206,18 @@ def create_bag_files(resource, fed_zone_home_path=''):
         '<ore:aggregates rdf:resource="%s"/>\n' % str(resource.metadata.type.url), '')
 
     # create resourcemap.xml and upload it to iRODS
-    from_file_name = os.path.join(bagit_path, 'resourcemap.xml')
+    from_file_name = os.path.join(temp_path, 'resourcemap.xml')
     with open(from_file_name, 'w') as out:
         out.write(xml_string)
-    to_file_name = os.path.join(resource.short_id, 'data', 'resourcemap.xml')
-    if fed_zone_home_path:
-        to_file_name = '{fed_zone_home_path}/{rel_path}'.format(
-            fed_zone_home_path=fed_zone_home_path,
-            rel_path=to_file_name)
-
+    if resource.resource_federation_path:
+        to_file_name = os.path.join(resource.resource_federation_path, 'data', 'resourcemap.xml')
+    else:
+        to_file_name = os.path.join(resource.short_id, 'data', 'resourcemap.xml')
```

Comment:
Same comment as above. The logic is not correct here and the root_path property can be used to simplify the if..else statement into one statement.